GVFS/GVFS.UnitTests/Git/GitAuthenticationTests.cs

```diff
@@ -168,17 +169,47 @@ public void TwoThreadsInterleavingFailuresShouldntStompASuccess()
             thread2Auth.ShouldEqual(thread1Auth, "The second thread stomp the first threads good auth string");
         }
 
+        [TestCase]
+        public void DontDoubleStoreExistingCredential()
+        {
+            MockTracer tracer = new MockTracer();
+            MockGitProcess gitProcess = this.GetGitProcess();
+
+            GitAuthentication dut = new GitAuthentication(gitProcess, "mock://repoUrl");
+            dut.TryInitializeAndRequireAuth(tracer, out _);
+
+            string authString;
+            dut.TryGetCredentials(tracer, out authString, out _).ShouldBeTrue();
+            dut.ConfirmCredentialsWorked(authString);

```

Comment:
You may want to validate the case where we send in a string other than what was retrieved from Get

Reply:
Added test.